urllib3/util/response.py

```diff
@@ -12,18 +12,21 @@ def is_fp_closed(obj):
         The file-like object to check.
     """
 
-    try:
+    isclosed = getattr(obj, 'isclosed', None)
+    if isclosed is not None:
+        # Check isclosed() for Python 3
+        return isclosed()
+
+    closed = getattr(obj, 'closed', None)
+    if closed is not None:
         # Check via the official file-like-object way.
-        return obj.closed
-    except AttributeError:
-        pass
+        return closed
 
-    try:
+    fp = getattr(obj, 'fp', False)
+    if fp is not False:
```

Comment:
This is a little ugly. Perhaps we should stick with the try/except method, but I was trying to adhere to the discussion @Lukasa and I had on a previous PR that getattr may be preferred to try/except.